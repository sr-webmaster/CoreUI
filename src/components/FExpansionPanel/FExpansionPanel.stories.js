import { MAIN } from '../../../.storybook/categories'
import FExpansionPanel from './FExpansionPanel.vue'

export default {
  title: `${MAIN}|FExpansionPanel`,
  id: 'FExpansionPanel',
  decorators: [
    () => ({
      template: `
        <v-container><story /></v-container>
      `
    })
  ]
}

/**
 * https://storybook.js.org/docs/guides/guide-vue/#step-4-write-your-stories
 */
export const WithTitleAndContent = () => ({
  data: () => ({
    items: [
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Photos', subtitle: 'Jan 9, 2014' },
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Recipes', subtitle: 'Jan 17, 2014' },
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Work', subtitle: 'Jan 28, 2014' }
    ]
  }),
  components: {
    FExpansionPanel
  },
  template: `
    <f-expansion-panel title="My Drive">
      <v-card-text>
        <v-list two-line>
          <v-list-tile
            v-for="item in items"
            :key="item.title"
            avatar
          >
            <v-list-tile-avatar>
              <v-icon :class="[item.iconClass]">{{ item.icon }}</v-icon>
            </v-list-tile-avatar>

            <v-list-tile-content>
              <v-list-tile-title>{{ item.title }}</v-list-tile-title>
              <v-list-tile-sub-title>{{ item.subtitle }}</v-list-tile-sub-title>
            </v-list-tile-content>

            <v-list-tile-action>
              <v-btn icon ripple>
                <v-icon color="grey lighten-1">info</v-icon>
              </v-btn>
            </v-list-tile-action>
          </v-list-tile>
        </v-list>
      </v-card-text>

      <v-card-actions>
        <v-btn
          flat
          color="orange"
        >
          Share
        </v-btn>
        <v-btn
          flat
          color="orange"
        >
          Explore
        </v-btn>
      </v-card-actions>
    </f-expansion-panel>
  `
})

export const ExpandFalse = () => ({
  data: () => ({
    items: [
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Photos', subtitle: 'Jan 9, 2014' },
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Recipes', subtitle: 'Jan 17, 2014' },
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Work', subtitle: 'Jan 28, 2014' }
    ]
  }),
  components: {
    FExpansionPanel
  },
  template: `
    <f-expansion-panel title="My Drive" :expand="false">
      <v-card-text>
        <v-list two-line>
          <v-list-tile
            v-for="item in items"
            :key="item.title"
            avatar
          >
            <v-list-tile-avatar>
              <v-icon :class="[item.iconClass]">{{ item.icon }}</v-icon>
            </v-list-tile-avatar>

            <v-list-tile-content>
              <v-list-tile-title>{{ item.title }}</v-list-tile-title>
              <v-list-tile-sub-title>{{ item.subtitle }}</v-list-tile-sub-title>
            </v-list-tile-content>

            <v-list-tile-action>
              <v-btn icon ripple>
                <v-icon color="grey lighten-1">info</v-icon>
              </v-btn>
            </v-list-tile-action>
          </v-list-tile>
        </v-list>
      </v-card-text>

      <v-card-actions>
        <v-btn
          flat
          color="orange"
        >
          Share
        </v-btn>
        <v-btn
          flat
          color="orange"
        >
          Explore
        </v-btn>
      </v-card-actions>
    </f-expansion-panel>
  `
})

export const NoTitle = () => ({
  data: () => ({
    items: [
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Photos', subtitle: 'Jan 9, 2014' },
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Recipes', subtitle: 'Jan 17, 2014' },
      { icon: 'folder', iconClass: 'grey lighten-1 white--text', title: 'Work', subtitle: 'Jan 28, 2014' }
    ]
  }),
  components: {
    FExpansionPanel
  },
  template: `
    <f-expansion-panel>
      <v-card-text>
        <v-list two-line>
          <v-list-tile
            v-for="item in items"
            :key="item.title"
            avatar
          >
            <v-list-tile-avatar>
              <v-icon :class="[item.iconClass]">{{ item.icon }}</v-icon>
            </v-list-tile-avatar>

            <v-list-tile-content>
              <v-list-tile-title>{{ item.title }}</v-list-tile-title>
              <v-list-tile-sub-title>{{ item.subtitle }}</v-list-tile-sub-title>
            </v-list-tile-content>

            <v-list-tile-action>
              <v-btn icon ripple>
                <v-icon color="grey lighten-1">info</v-icon>
              </v-btn>
            </v-list-tile-action>
          </v-list-tile>
        </v-list>
      </v-card-text>

      <v-card-actions>
        <v-btn
          flat
          color="orange"
        >
          Share
        </v-btn>
        <v-btn
          flat
          color="orange"
        >
          Explore
        </v-btn>
      </v-card-actions>
    </f-expansion-panel>
  `
})
