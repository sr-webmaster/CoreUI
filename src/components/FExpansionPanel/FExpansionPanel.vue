<template>
  <v-expansion-panel
    expand
    focusable
    :value="[expand]"
  >
    <v-expansion-panel-content>
      <template v-slot:header>
        <h3>{{ title }}</h3>
      </template>
      <v-divider />
      <v-card :class="cardClass">
        <slot />
      </v-card>
    </v-expansion-panel-content>
  </v-expansion-panel>
</template>

<script>
export default {
  props: {
    cardClass: {
      type: String,
      default: ''
    },
    title: {
      type: String,
      default: ''
    },
    expand: {
      type: Boolean,
      default: true
    }
  }
}
</script>
