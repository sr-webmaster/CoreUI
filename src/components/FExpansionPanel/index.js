import Component from './FExpansionPanel'
export default Component
