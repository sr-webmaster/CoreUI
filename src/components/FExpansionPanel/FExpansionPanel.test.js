import { mount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import { WithTitleAndContent, NoTitle, ExpandFalse } from './FExpansionPanel.stories'

describe('FExpansionPanel', () => {
  describe('Computed', () => {
    // Using shallowMount test the computed properties
  })

  describe('Methods', () => {
    // Using shallowMount test the methods
  })

  describe('Visuals', () => {
    test('WithTitleAndContent', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(WithTitleAndContent(), {
        localVue: createLocalVue().localVue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
    test('NoTitle', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(NoTitle(), {
        localVue: createLocalVue().localVue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
    test('ExpandFalse', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(ExpandFalse(), {
        localVue: createLocalVue().localVue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
