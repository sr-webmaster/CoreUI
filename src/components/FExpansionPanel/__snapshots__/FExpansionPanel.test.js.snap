// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FExpansionPanel Visuals ExpandFalse 1`] = `
<ul
  class="v-expansion-panel v-expansion-panel--focusable theme--light"
>
  <li
    class="v-expansion-panel__container"
    tabindex="0"
  >
    <div
      class="v-expansion-panel__header"
    >
      <h3>
        My Drive
      </h3>
      <div
        class="v-expansion-panel__header__icon"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light"
        >
          keyboard_arrow_down
        </i>
      </div>
    </div>
    <div
      class="v-expansion-panel__body"
      style="display: none;"
    >
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <div
            class="v-list v-list--two-line theme--light"
            role="list"
          >
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Photos
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 9, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Recipes
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 17, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Work
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 28, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <button
            class="v-btn v-btn--flat theme--light orange--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Share
        
            </div>
          </button>
           
          <button
            class="v-btn v-btn--flat theme--light orange--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Explore
        
            </div>
          </button>
        </div>
      </div>
    </div>
  </li>
</ul>
`;

exports[`FExpansionPanel Visuals NoTitle 1`] = `
<ul
  class="v-expansion-panel v-expansion-panel--focusable theme--light"
>
  <li
    class="v-expansion-panel__container"
    tabindex="0"
  >
    <div
      class="v-expansion-panel__header"
    >
      <h3>
        
      </h3>
      <div
        class="v-expansion-panel__header__icon"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light"
        >
          keyboard_arrow_down
        </i>
      </div>
    </div>
    <div
      class="v-expansion-panel__body"
      style="display: none;"
    >
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <div
            class="v-list v-list--two-line theme--light"
            role="list"
          >
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Photos
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 9, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Recipes
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 17, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Work
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 28, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <button
            class="v-btn v-btn--flat theme--light orange--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Share
        
            </div>
          </button>
           
          <button
            class="v-btn v-btn--flat theme--light orange--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Explore
        
            </div>
          </button>
        </div>
      </div>
    </div>
  </li>
</ul>
`;

exports[`FExpansionPanel Visuals WithTitleAndContent 1`] = `
<ul
  class="v-expansion-panel v-expansion-panel--focusable theme--light"
>
  <li
    class="v-expansion-panel__container"
    tabindex="0"
  >
    <div
      class="v-expansion-panel__header"
    >
      <h3>
        My Drive
      </h3>
      <div
        class="v-expansion-panel__header__icon"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light"
        >
          keyboard_arrow_down
        </i>
      </div>
    </div>
    <div
      class="v-expansion-panel__body"
      style="display: none;"
    >
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="v-card v-sheet theme--light"
      >
        <div
          class="v-card__text"
        >
          <div
            class="v-list v-list--two-line theme--light"
            role="list"
          >
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Photos
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 9, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Recipes
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 17, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              role="listitem"
            >
              <div
                class="v-list__tile v-list__tile--avatar theme--light"
              >
                <div
                  class="v-list__tile__avatar"
                >
                  <div
                    class="v-avatar"
                    style="height: 40px; width: 40px;"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon material-icons theme--light grey lighten-1 white--text"
                    >
                      folder
                    </i>
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__content"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Work
                  </div>
                   
                  <div
                    class="v-list__tile__sub-title"
                  >
                    Jan 28, 2014
                  </div>
                </div>
                 
                <div
                  class="v-list__tile__action"
                >
                  <button
                    class="v-btn v-btn--icon theme--light"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light grey--text text--lighten-1"
                      >
                        info
                      </i>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-card__actions"
        >
          <button
            class="v-btn v-btn--flat theme--light orange--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Share
        
            </div>
          </button>
           
          <button
            class="v-btn v-btn--flat theme--light orange--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Explore
        
            </div>
          </button>
        </div>
      </div>
    </div>
  </li>
</ul>
`;
